### PR TITLE
Development updates

### PR DESCRIPTION
**Pull request by gaissmai**
## Summary by CodeRabbit

- **New Features**
  - Added routing-table aggregation for IPv4 and IPv6 prefixes.
  - Combine overlapping or adjacent prefixes into minimal covering CIDR blocks.
  - Recalculate table sizes after aggregation.

- **Bug Fixes**
  - Improved recursive aggregation across nested routing entries.
  - Preserved address-membership behavior while removing redundant prefixes.
  - Aggregation is idempotent and supports mixed IPv4/IPv6 tables.

### CODE REVIEW

**Comment by gaissmai** [comment]
@coderabbit pause

**Comment by gaissmai** [comment]
@greptileai schreibe bitte TDT tests für Lite.Aggregate die die bisher gemachten Fehler abtesten und darüber hinaus möglichst alle corner cases.

**Comment by gaissmai** [comment]
@coderabbit review full

File: internal/nodes/nodebasics.go [hunk 1]
```diff
@@ -332,3 +332,35 @@ func (l *FringeNode[V]) CloneFringe(cloneFn func(V) V) *FringeNode[V] {
...
+// provided they form a contiguous, power-of-two aligned pair.
+//
+// Invariants:
+//   - Both prefixes a and b must be valid (pfx.IsValid() == true).
+//   - Both prefixes must belong to the same address family (IPv4 or IPv6).
+//
+// It returns the Supernet prefix with (bits - 1) and true if the prefixes can be merged;
+// otherwise, it returns an invalid netip.Prefix{} and false.
+func Supernet(a, b netip.Prefix) (netip.Prefix, bool) {
+	if a == b {
```
[the remark this answers is not in the document]
Comment (gaissmai) [reply internal/nodes/nodebasics.go]: @coderabbit die invariante ist, dass nur canonicalized prefixes im trie sind

File: internal/nodes/lite.go [hunk 2]
```diff
@@ -141,3 +144,186 @@ func (n *LiteNode[V]) CloneFlat(_ func(V) V) *LiteNode[V] {
...
+			}
+
+			lastIdx = idx
+		}
+	}
+
+	// 6. Recursive Descent: Top-down compression of child LiteNodes.
+	for _, anyKid := range n.Children.Items {
+		if kid, ok := anyKid.(*LiteNode[V]); ok {
+			modified += kid.Aggregate()
```
[the remark this answers is not in the document]
Comment (gaissmai) [reply internal/nodes/lite.go]: @coderabbit sehr gut gemacht, das hatte ich übersehen!

[the remark this answers is not in the document]
Comment (gaissmai) [reply internal/nodes/lite.go]: @coderabbit mach mir dazu bitte einen Vorschlag wie ich das fixen könnte

File: internal/nodes/lite.go [hunk 3]
```diff
@@ -141,3 +144,128 @@ func (n *LiteNode[V]) CloneFlat(_ func(V) V) *LiteNode[V] {
...
+		covered := n.Children.Intersection(&allot.FringeRoutesLookupTbl[idx])
+		for addr := range covered.All() {
+			n.DeleteChild(addr)
+			modified++
+		}
+	}
+
+	// 3. Recursive Descent: Top-down compression of child LiteNodes.
+	for i, anyKid := range n.Children.Items {
+		if kid, ok := anyKid.(*LiteNode[V]); ok {
```
[the remark this answers is not in the document]
Comment (gaissmai) [reply internal/nodes/lite.go:197]: @greptileai das sind beides fringes:
```text
$ go test -run=My -v
=== RUN   TestMy
    my_test.go:23: ▼
        ├─ 10.0.0.0/24
        └─ 10.0.1.0/24
        
        ### IPv4: size(2), subnodes(3), prefixes(0), leaves(0), fringes(2)
        [PATH] depth:  0 path: [] / 0
        octets(#1): [10]
        childs(#1): 10
        
        .[PATH] depth:  1 path: [10] / 8
        .octets(#1): [0]
        .childs(#1): 0
        
        ..[STOP] depth:  2 path: [10.0] / 16
        ..octets(#2): [0 1]
        ..fringe(#2): 0:{10.0.0.0/24} 1:{10.0.1.0/24}
        
    my_test.go:30: ▼
        └─ 10.0.0.0/23
        
        ### IPv4: size(1), subnodes(3), prefixes(1), leaves(0), fringes(0)
        [PATH] depth:  0 path: [] / 0
        octets(#1): [10]
        childs(#1): 10
        
        .[PATH] depth:  1 path: [10] / 8
        .octets(#1): [0]
        .childs(#1): 0
        
        ..[STOP] depth:  2 path: [10.0] / 16
        ..indexs(#1): [128]
        ..prefxs(#1): 10.0.0.0/23
        
--- PASS: TestMy (0.00s)
PASS
```

**Comment by gaissmai** [comment]
@greptileai das sind keine adjacent siblings in deinem Testcase:
```go
		{
			name:  "adjacent siblings ipv6",
			input: []string{"2001:db8::/57", "2001:db8:0:0:8000::/57"},
			want:  []string{"2001:db8::/56"},
		},
```

**Comment by gaissmai** [comment]
@coderabbit review
